ros/src/tl_detector/tl_detector.py

```diff
@@ -203,41 +206,59 @@ def process_traffic_lights(self):
 
         """
         light = None
+        car_position = 0
 
         # List of positions that correspond to the line to stop in front of for a given intersection
         #stop_line_positions = self.config['stop_line_positions']
 	
         ntl_wp = -1
         gt_ntl_state =TrafficLight.UNKNOWN
-	ntl_state =TrafficLight.UNKNOWN
+        ntl_state =TrafficLight.UNKNOWN
+
+        if self.light_classifier is not None: 
+            if not self.busy:
+                self.busy = True
+                ntl_state = self.get_light_state(light)
+                self.previous_light_state = ntl_state
```

Comment:
I thought about this- but was worried if we will hit a racing condition.

Reply:
I thought about this- but was worried if we will hit a racing condition.

Reply:
Not sure, but it seems ok to me.  I'm concerned that if we don't, theoretically if the images dropped out and stop being received it will stay latched on the last image and the classifier will keep detecting the same thing so the state will satisfy the 3 state_count threshold and the light state will get stuck at what the last image showed.